packages/hoppscotch-common/src/services/spotlight/searchers/environment.searcher.ts

```diff
@@ -370,10 +440,23 @@ export class SwitchEnvSpotlightSearcherService
   }
 
   onResultSelect(result: SpotlightSearcherResult): void {
-    const selectedEnvIndex = Number(result.id.split("-")[1])
-    setSelectedEnvironmentIndex({
-      type: "MY_ENV",
-      index: selectedEnvIndex,
-    })
+    if (result.id.startsWith("team-environment")) {
```

Comment:
Instead of matching the ID, can we introduce a field to infer the environment type and comparing against that?

https://github.com/hoppscotch/hoppscotch/blob/286fcd2bb08a84f027b10308d1e18da368f95ebf/packages/hoppscotch-common/src/newstore/environments.ts#L9-L17